weave-js/src/components/ToggleButtonGroup/ToggleButtonGroup.tsx

```diff
@@ -0,0 +1,65 @@
+import * as ToggleGroup from '@radix-ui/react-toggle-group';
+import React from 'react';
+import {twMerge} from 'tailwind-merge';
+
+import {Button} from '../Button';
+import {IconName} from '../Icon';
+import {Tailwind} from '../Tailwind';
+import {ToggleButtonGroupSize} from './types';
+
+export type ToggleButtonGroupProps = {
+  options: string[];
+  value: string;
+  size: ToggleButtonGroupSize;
+  isDisabled?: boolean;
+  icons?: Array<IconName | undefined>;
+  onValueChange: (value: string) => void;
+};
+
+export const ToggleButtonGroup = React.forwardRef<
+  HTMLDivElement,
+  ToggleButtonGroupProps
+>(({options, value, size, isDisabled = false, icons, onValueChange}, ref) => {
+  if (options.length < 2) {
+    console.error('ToggleButtonGroup requires at least two options.');
+  }
+
+  const handleValueChange = (newValue: string) => {
+    if (newValue !== value) {
+      onValueChange(newValue);
+    }
+  };
+  return (
+    <Tailwind>
+      <ToggleGroup.Root
+        type="single" // supports single selection only
+        value={value}
+        onValueChange={handleValueChange}
+        className="flex gap-px"
+        ref={ref}>
+        {options.map((option, index) => (
+          <ToggleGroup.Item key={option} value={option} disabled={isDisabled}>
+            <Button
+              icon={icons?.[index]}
```

Comment:
suggestion - maybe we should also outright disable the option for the one that is already selected. keep the check in your handler (newValue !== value) but add another layer of safety by not allowing the user to even click the option that is already toggled)

Reply:
good idea, i'll update!